grammar/argextend.mlp

```diff
@@ -185,6 +185,7 @@ let declare_vernac_argument loc s pr cl =
    [ <:str_item<
       value ($lid:"wit_"^s$ : Genarg.genarg_type 'a unit unit) =
         Genarg.create_arg $se$ >>;
+     <:str_item< Geninterp.register_val0 $wit$ None >>;
```

Comment:
This is useless, because vernacular arguments have no toplevel values, and thus this will never be used.

Reply:
This is because `declare_extra_genarg_pprule` now calls `register_print0`. See the commit log below for a possible (more complicated) alternative if you think it is better:

    Registering vernac arguments as values to satisfy printer declarations.
    
    Printer declarations (in declare_extra_genarg_pprule) shall soon
    register printers for values, so we have to declare vernac arguments
    as values.
    
    This is a bit overkill because we actually give dummy "glob" and "top"
    printers for vernac arguments, so it should be probably more
    consistent to have a version of declare_extra_genarg_pprule which
    declares only a "raw" printer, but that would require more changes in
    the tables for printer registration, so we go instead the shortest
    path.

Reply:
Ah, yep, I had overlooked that one. Well, I think a middle ground solution would be to separate the registration of the Val.t printer from the genarg one. In practice, most of them are given by ARGUMENT EXTEND, so it's just a matter of adapting that one.

Also, currently the top part of a genarg is mostly useless. It is only used for phantom typing the evaluation of a generic argument, in a way which actually is not quite sound with respect to the dynamic typing nature of Ltac. (IIRC there is also some legacy use for printing tactic traces, but that one is dead wrong and we should take advantage of your PR to fix it.)